### PR TITLE
problem_solving_methods

### PR DESCRIPTION
**Pull request by afonsobspinto**
### In this PR I have done the following:
 * add heuristics
 * add book 
 * fix minimax broken link
 * add branch and bound
 * add hill climbing
 * add simulated annealing

### CODE REVIEW

**Comment by afonsobspinto** [comment]
sorry for the horrible pictures I got. :pray: to be changed

File: summary/problem_solving_methods.md [hunk 1]
```diff
@@ -65,6 +61,15 @@ TODO
                 *   optimally efficient: no better algorithm with same knowledge
         2.  **Heuristic functions**
             *   Relax the demands of the problem
+            1. Admissible
```
Comment (msramalho) [inline summary/problem_solving_methods.md:64]: This distinction between admissible and consistent is very good

File: summary/problem_solving_methods.md [hunk 2]
```diff
@@ -192,7 +225,7 @@ Problem from figure 2.2 in [Intelligent Systems for Engineering: A Knowledge-bas
 
 **Question:** Given a tree of `r = 2` (branching factor) and  `m = 2` (tree depth), with node values of 5,3,2,7, what are the values of α and β for each tree node and specify if all the tree nodes are visited.
 
-**Answer:** node with value `7` is cut and the values of α and β can be visualized [here](http://proof.github.io/minimax/#tree=KCg1LDMpLCgyLDcpKQ%3D%3D).
```
Comment (msramalho) [inline summary/problem_solving_methods.md:195]: I tought it was dead!!

**Review by msramalho — APPROVED** [review]
Awesome work, truly

**Pull request merged**